eppoclient/assignmentlogger_test.go

```diff
@@ -0,0 +1,79 @@
+package eppoclient
+
+import (
+	"encoding/json"
+	"reflect"
+	"testing"
+)
+
+// TestAssignmentEventSerialization tests the JSON serialization and deserialization of AssignmentEvent
+func TestAssignmentEventSerialization(t *testing.T) {
+	// Create a test case with each type
+	testCases := []AssignmentEvent{
+		// empty subject attributes
+		{
+			Experiment:  "testExperiment",
+			FeatureFlag: "testFeatureFlag",
+			Allocation:  "testAllocation",
+			Variation:   String("testVariation"),
+			Subject:     "testSubject",
+			Timestamp:   "testTimestamp",
+		},
+		{
+			Experiment:        "testExperiment",
+			FeatureFlag:       "testFeatureFlag",
+			Allocation:        "testAllocation",
+			Variation:         Bool(true),
+			Subject:           "testSubject",
+			Timestamp:         "testTimestamp",
+			SubjectAttributes: dictionary{"testKey": "testValue"},
```

Comment:
We have a known failure that when a `SubjectAttributes` value is instantiated as anything but a string, it gets serialized only into a string.

```
SubjectAttributes: dictionary{"testKey": 5}
SubjectAttributes: dictionary{"testKey": "5"}
```

This is important to figure out in the case where users provide the fields and they are sent to the assignment logger in a string type.

Reply:
🤔 Strange